sql/core/src/main/java/org/apache/spark/sql/vectorized/ArrowColumnVector.java

```diff
@@ -28,21 +28,14 @@
 /**
  * A column vector backed by Apache Arrow.
  */
-public final class ArrowColumnVector extends ColumnVector {
+public final class ArrowColumnVector implements ColumnVector {
 
   private final ArrowVectorAccessor accessor;
   private ArrowColumnVector[] childColumns;
 
-  private void ensureAccessible(int index) {
```

Comment:
`ColumnVector` is a performance critical place, we don't need index checking here, like other column vector implementations.